Entities/SoundContainer.h

```diff
@@ -0,0 +1,272 @@
+		int GetLoopSetting() const { return m_Loops; }
+
+		/// <summary>
+		/// Sets the looping setting of this SoundContainer. 
+		/// 0 means the sound is set to only play once. -1 means it loops indefinitely.
+		/// </summary>
+		/// <param name="loops">An int with the loop count.</param>
+		void SetLoopSetting(int loops);
+
+		/// <summary>
+		/// Gets the current playback priority.
+		/// </summary>
+		/// <returns>The playback priority.</returns>
+		int GetPriority() const { return m_Priority; }
+
+		/// <summary>
+		/// Sets the current playback priority. Higher value will make this more likely to make it into mixing on playback.
+		/// </summary>
+		/// <param name="priority">The new priority. See AudioMan::PRIORITY_* enumeration.</param>
+		void SetPriority(int priority) { m_Priority = priority; }
+
+		/// <summary>
+		/// Gets whether the sounds in this SoundContainer are affected by global pitch changes or not.
+		/// </summary>
+		/// <returns>Whether or not the Sounds in this SoundContainer affected by global pitch changes.</returns>
+		bool IsAffectedByGlobalPitch() const { return m_AffectedByGlobalPitch; }
+
+		/// <summary>
+		/// Sets whether the sounds in this SoundContainer are affected by global pitch changes or not.
+		/// </summary>
+		/// <param name="pitched">Whether the sounds in this SoundContainer should be affected by global pitch or not.</param>
+		void SetAffectedByGlobalPitch(bool affectedByGlobalPitch) { m_AffectedByGlobalPitch = affectedByGlobalPitch; }
+
+		/// <summary>
+		/// Gets the hash for this SoundContainer, used for multiplayer data
+		/// </summary>
+		/// <returns>The hash for this SoundContainer</returns>
+		size_t GetHash() { return m_Hash; }
+#pragma endregion
+
+#pragma region Playback Controls
+		/// <summary>
+		/// Plays the next sound of this SoundContainer with default attenuation for all players.
+		/// </summary>
+		/// <returns>Whether this SoundContainer successfully started playing on any channels.</returns>
+		bool Play() { return Play(0, -1); }
+
+		/// <summary>
+		/// Plays the next sound of this SoundContainer with the given attenuation for all players.
+		/// </summary>
+		/// <param name="attenuation">How much distance attenuation to apply to the SoundContainer.</param>
+		/// <returns>Whether this SoundContainer successfully started playing on any channels.</returns>
+		bool Play(float attenuation) { return Play(attenuation, -1); }
+
+		/// <summary>
+		/// Plays the next sound of this SoundContainer with the given attenuation for a specific player.
+		/// </summary>
+		/// <param name="attenuation">How much distance attenuation to apply to the SoundContainer.</param>
+		/// <param name="player">Player to start playback of this SoundContainer for.</param>
+		/// <returns>Whether this SoundContainer successfully started playing on any channels.</returns>
+		bool Play(float attenuation, int player) { return HasAnySounds() ? g_AudioMan.PlaySound(this, player, attenuation) : false; }
```

Comment:
Consider placing the `player` parameter first (or second where a pointer to a sound is needed) for convenience.
Applies for everything.

Reply:
I spent a while thinking on this and changing it around. My reasoning is put the most important, necessary things first and attenuation will be one of those as soon as we have 3d sound (and kind of is right now).

With 3d sound every sound call will have to put in a position but not all of them will have to put in a player. So, the ordering goes `filePath/soundContainer, position, player, priorityHandling, pitchHandling`.